testing/fil_token_integration/actors/basic_token_actor/src/lib.rs

```diff
@@ -0,0 +1,309 @@
+    std::panic::set_hook(Box::new(|info| {
+        sdk::vm::abort(ExitCode::USR_ASSERTION_FAILED.value(), Some(&format!("{}", info)))
+    }));
+
+    let method_num = sdk::message::method_number();
+
+    match method_num {
+        // Actor constructor
+        1 => constructor(),
+
+        // Standard token interface
+        rest => {
+            let root_cid = sdk::sself::root().unwrap();
+
+            let bs = Blockstore::default();
+            let mut token_state = Token::<_, FvmMessenger>::load_state(&bs, &root_cid).unwrap();
+
+            let mut token_actor =
+                BasicToken { util: Token::wrap(bs, FvmMessenger::default(), 1, &mut token_state) };
+
+            // Method numbers calculated via fvm_dispatch_tools using CamelCase names derived from
+            // the corresponding FRCXXXToken trait methods.
+            match rest {
+                4244593718 => {
+                    // Name
+                    let name = token_actor.name();
+                    return_ipld(&name).unwrap()
+                }
+                3551111368 => {
+                    // Symbol
+                    let symbol = token_actor.symbol();
+                    return_ipld(&symbol).unwrap()
+                }
+                2511420746 => {
+                    // TotalSupply
+                    let total_supply = token_actor.total_supply();
+                    return_ipld(&BigIntDe(total_supply)).unwrap()
+                }
+                1568445334 => {
+                    //BalanceOf
+                    let params = deserialize_params(params);
+                    let res = token_actor.balance_of(params).unwrap();
+                    return_ipld(&BigIntDe(res)).unwrap()
+                }
+                2804639308 => {
+                    // Allowance
+                    let params = deserialize_params(params);
+                    let res = token_actor.allowance(params).unwrap();
+                    return_ipld(&BigIntDe(res)).unwrap()
+                }
+                991449938 => {
+                    // IncreaseAllowance
+                    let params = deserialize_params(params);
+                    let res = token_actor.increase_allowance(params).unwrap();
+                    token_actor.util.flush().unwrap();
+                    return_ipld(&BigIntDe(res)).unwrap()
+                }
+                4218751446 => {
+                    // DecreaseAllowance
+                    let params = deserialize_params(params);
+                    let res = token_actor.decrease_allowance(params).unwrap();
+                    token_actor.util.flush().unwrap();
+                    return_ipld(&BigIntDe(res)).unwrap()
+                }
+                1691518633 => {
+                    // RevokeAllowance
+                    let params = deserialize_params(params);
+                    token_actor.revoke_allowance(params).unwrap();
+                    token_actor.util.flush().unwrap();
```

Comment:
Should every method call that `flush`es also `set_root` afterwards?

Reply:
ah, yes I think so